### PR TITLE
AP_Volz_Protocol scaling bugfix

### PR DESCRIPTION
**Pull request by lazarovtsvetozar**
AP_Volz_Protocol: scaling bugfix

### CODE REVIEW

File: libraries/AP_Volz_Protocol/AP_Volz_Protocol.h [hunk 1]
```diff
@@ -48,6 +48,9 @@
 #define VOLZ_EXTENDED_POSITION_CENTER 		0x0800	// Extended Position Data Format defines 0 as 0x0800 - decimal 2048
 #define VOLZ_EXTENDED_POSITION_MAX 			0x0F80	// Extended Position Data Format defines +100 as 0x0F80 decimal 3968 -> full range decimal 3840
 
+#define PWM_POSITION_MIN 					1000
```
Comment (rmackay9) [inline libraries/AP_Volz_Protocol/AP_Volz_Protocol.h:51]: We should probably name these definitions starting with VOLZ_.  So perhaps VOLZ_PWM_MIN and VOLZ_PWM_MAX.  Aleternatively move them to the .cpp file to keep their scope small but I think from a readers point of view it is nice to have all the definitions together.

File: libraries/AP_Volz_Protocol/AP_Volz_Protocol.cpp [hunk 2]
```diff
@@ -73,15 +73,16 @@ void AP_Volz_Protocol::update()
...
                 value = 0;
             } else {
-                value = c->get_output_pwm() - c->get_output_min();
+                value = c->get_output_pwm() - PWM_POSITION_MIN;
             }
 
             // scale the PWM value to Volz value
-            value = value + VOLZ_EXTENDED_POSITION_MIN;
-            value = value * VOLZ_SCALE_VALUE / (c->get_output_max() - c->get_output_min());
+
```
Comment (rmackay9) [inline libraries/AP_Volz_Protocol/AP_Volz_Protocol.cpp:83]: a small white space issue here.  We should remove the blank line under the comment and indent the lines to the same amount.

**Review by rmackay9 — CHANGES_REQUESTED** [review]
It's great to see that someone is actually using this library so that we know that it's being used.

A few small change requests:

- remove the change to the mavlink submodule
- the commit should be changed to "AP_Volz_Protocol: scaling bugfix".  Please note that I'm talking about the commit itself, not the PR title.
- fixup the whitespace issue
- add VOLZ_ to the front of the new definitions

Also if you could confirm what testing you've done that would be helpful in getting this merged, thanks!

**Comment by lazarovtsvetozar** [comment]
Tests done so far: 
1. Scaling correction: 
We tested with cube black autopilot connected to Mission Planner. We configured one of the VOLZ Channels (servo3) as a Throttle. Then connected to an oscilloscope with which we translate the message from the autopilot and confirm that the scaling method was working after the change.
2. Hard-coded VOLZ_PWM_MIN / VOLZ_PWM_MAX
In this case when we change the lower value of the SERVO3_MIN = 1200 (for example) the minimum value of the VOLZ_Protocol shifts with 20%, as we have hard coded values of 1000 for minimum scaling factor and 2000 for maximum scaling factor for calculating the Volz limits.

**Comment by peterbarker** [comment]
@lazarovtsvetozar this one needs a rebase.

I'd do it for you, but it looks like you've PR'd from your master branch rather than a git topic branch.

Could you rebase this, please?

**Comment by lazarovtsvetozar** [comment]
> 
> 
> @lazarovtsvetozar this one needs a rebase.
> 
> I'd do it for you, but it looks like you've PR'd from your master branch rather than a git topic branch.
> 
> Could you rebase this, please?

Hi Peter (@peterbarker), 
Well I`m kind of new here in git hub and I`m not so well knowledgeable of the terminology of the Git Hub community. Could you please give a hint how to do it, because I thought I was doing the right thing but actually I did it wrong.  Or if you have the time to do it instead of me it would be great ?! 
Thank you in advance, I will appreciate if this change happen in the near future!
Have a nice day !

**Comment by rmackay9** [comment]
I think this change is OK although I don't have any servos to test with.

**Comment by peterbarker** [comment]
@lazarovtsvetozar I've rebased this one for you.

Generally you don't want to make changes to your master branch - you create a git topic branch (based on master) and work on that, push that branch up to github and create a PR from that.  google will help you find more information on topic branches.

git is ridiculously powerful - I suggest you invest some time in getting to know it :-)

**Review by rmackay9 — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks @lazarovtsvetozar

**Comment by lazarovtsvetozar** [comment]
Hi Peter (@peterbarker), 
Let me thank you for the quick respond! I will do my best with git hub as soon as it is possible to me! 
I would like to ask one more question even though it may not be relevant to be ask here but we struggle with [this](https://github.com/ArduPilot/ardupilot/issues/15563) 
Is there by any chance this to be changed ?
Wish you all the best!
Thank you in advanced!

**Pull request merged**